lib/Dimension.js

```diff
@@ -5,32 +5,34 @@ const ns = require('./namespaces')
 const { toTerm } = require('./utils')
 
 class Dimension extends Node {
-  constructor ({ parent, term, dataset, graph, aggregate, as, join, path, source }) {
+  constructor ({ parent, term, dataset, graph, aggregate, as, join, path, source, createTriples = true }) {
```

Comment:
Added a flag 'createTriples' to distinguish when creating objects from data or not. When creating the object from data, the quads will be already there.